shared/app/tab-bar/index.render.js.flow

```diff
@@ -1,12 +1,13 @@
 // @flow
 import React, {Component} from 'react'
 import type {Tab} from '../../constants/tabs'
+import type {Map} from 'immutable'
 
 export type Props = {
   onTabClick: (tab: Tab) => void,
   selectedTab: Tab,
   username?: string,
-  badgeNumbers: {[key: Tab]: number},
+  badgeNumbers: Map<Tab, number>,
```

Comment:
The alternative would be doing a `shouldComponentUpdate` with a deep compare when the component updates. Wouldn't it be preferable to pass an immutable object down instead of converting to object (copying) for no reason and then doing the extra compare?